### ISSUE TITLE
Crash after logging in preceded by a failed login

### ISSUE THREAD

**Issue by MTRNord** [element-hq/hydrogen-web#184]
**Describe the bug**
This shows up:
![grafik](https://user-images.githubusercontent.com/1374914/98010408-f2f16a00-1df6-11eb-8912-a7b57d70b62d.png)

**To Reproduce**
1. Login with wrong server first (for example `matrix.ffslfl.net` which is a dead server which gives still a http response) (Unsure if contributing here)
1.5. (You might run into https://github.com/vector-im/hydrogen-web/issues/152 )
2. Login with correct data
3. The above error happens after first sync.

**Expected behavior**
UI should look normal

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Firefox
 - Version: 82.0.2 (64-Bit)

**Additional context**
I think it might be that the first sync failed causing this.

**Comment by bwindels** [comment]
This was on 0.1.24 btw

**Comment by MTRNord** [comment]
Whoopes missinterpreted the Version question :)

**Comment by bwindels** [comment]
> Whoopes missinterpreted the Version question :)

Ah :smile: No, that's just the default github template.

**Comment by MTRNord** [comment]
[console-export-2020-11-3_17-30-5.txt](https://github.com/vector-im/hydrogen-web/files/5482637/console-export-2020-11-3_17-30-5.txt)

**Comment by erdnaxeli** [comment]
I just got this issue. It was my first login on this browser (firefox 82.0.2, debian sid), and my credentials were correct, as I received a notification on element asking me to verify the new connection from hydrogen.

```
Something went wrong…
vm is undefined

This occurred while running view@https://hydrogen.element.io/hydrogen-2492926000.js:2034:20.

render@https://hydrogen.element.io/hydrogen-2492926000.js:2110:18
mount@https://hydrogen.element.io/hydrogen-2492926000.js:1870:31
loadList@https://hydrogen.element.io/hydrogen-2492926000.js:1756:40
mount@https://hydrogen.element.io/hydrogen-2492926000.js:1716:14
view@https://hydrogen.element.io/hydrogen-2492926000.js:2032:25
render@https://hydrogen.element.io/hydrogen-2492926000.js:2188:15
mount@https://hydrogen.element.io/hydrogen-2492926000.js:1870:31
view@https://hydrogen.element.io/hydrogen-2492926000.js:2032:25
render@https://hydrogen.element.io/hydrogen-2492926000.js:2698:15
mount@https://hydrogen.element.io/hydrogen-2492926000.js:1870:31
view@https://hydrogen.element.io/hydrogen-2492926000.js:2032:25
mapView/<@https://hydrogen.element.io/hydrogen-2492926000.js:2054:29
binding@https://hydrogen.element.io/hydrogen-2492926000.js:2002:43
update@https://hydrogen.element.io/hydrogen-2492926000.js:1900:17
_updateFromValue@https://hydrogen.element.io/hydrogen-2492926000.js:1894:14
emit@https://hydrogen.element.io/hydrogen-2492926000.js:4523:18
emitChange@https://hydrogen.element.io/hydrogen-2492926000.js:10315:18
_setSection@https://hydrogen.element.io/hydrogen-2492926000.js:12778:14
_showSession@https://hydrogen.element.io/hydrogen-2492926000.js:12734:14
_applyNavigation@https://hydrogen.element.io/hydrogen-2492926000.js:12683:26
load/<@https://hydrogen.element.io/hydrogen-2492926000.js:12663:74
emit@https://hydrogen.element.io/hydrogen-2492926000.js:3149:14
emitIfChanged@https://hydrogen.element.io/hydrogen-2492926000.js:11731:18
applyPath@https://hydrogen.element.io/hydrogen-2492926000.js:11627:42
push@https://hydrogen.element.io/hydrogen-2492926000.js:11618:21
ready@https://hydrogen.element.io/hydrogen-2492926000.js:12728:37
ready@https://hydrogen.element.io/hydrogen-2492926000.js:12481:22
start@https://hydrogen.element.io/hydrogen-2492926000.js:12386:22
```

I see this error in the console:
```
Failed to load ‘https://cloud.cervoi.se/_matrix/client/r0/sync?timeout=0&filter=309&_cacheBuster=4816286513083069’. A ServiceWorker intercepted the request and encountered an unexpected error. sw.js:73:10
```

**Comment by bwindels** [comment]
Was this on an account that hadn't joined any rooms yet? If so, this was fixed by https://github.com/vector-im/hydrogen-web/commit/b27f6a067f19892a79338526193c7856e2f003ce

**Comment by erdnaxeli** [comment]
For me this was not the case, the accounted had joined several rooms.

**Comment by MTRNord** [comment]
For me the dead one account (first one) obviously had no rooms as it was dead ^^ The second one had plenty of rooms

**Comment by bwindels** [comment]
I wonder if an extension is throwing a wrench in the cogs... Any extensions you both might have in common?

**Comment by bwindels** [comment]
Hmm, looking at the code, it looks like the sync request threw a [DOMException with name `AbortError`](https://developer.mozilla.org/en-US/docs/Web/API/DOMException#exception-AbortError). This error is ignored by the sync as an error with the same name is used when we close the session, so it proceeds to show the session screen, which threw with an empty room list before https://github.com/vector-im/hydrogen-web/commit/b27f6a067f19892a79338526193c7856e2f003ce

Question is, what is throwing the AbortError?

**Comment by MTRNord** [comment]
The only network intercepting addons I do use are uBlockOrigin and PrivaccyBadger

**Comment by bwindels** [comment]
> The only network intercepting addons I do use are uBlockOrigin and PrivaccyBadger

If you can reproduce the failure, could you retry without those enabled please?

**Comment by MTRNord** [comment]
Sure i will try later

**Comment by MTRNord** [comment]
Hm not sure but it seems to be the uBlockOrigin addon. however it seems to be flaky anyway the login. had a corrupted session in between. So I am unsure about my test result. Will try later again

**Comment by bwindels** [comment]
Hmm, not sure it's that. When the extension blocks a request, it seems to give a `TypeError` (i.e. offline) rather than an `AbortError`. The network tab should also tell you it was blocked by an extension:
![image](https://user-images.githubusercontent.com/274386/98280920-fec66300-1f93-11eb-8e13-ba587c6250ac.png)

**Comment by bwindels** [comment]
`A ServiceWorker intercepted the request and encountered an unexpected error.` corresponds to the firefox translation string `InterceptionFailedWithURL`, but there's very little information to be found when this is reported...

**Comment by bwindels** [comment]
The crash should be fixed in b27f6a0 and the fact that you got through to the session view even though sync failed is fixed in 134f018 and b8370d5, and I split off an issue for the firefox service worker error that might cause the AbortError in #187, so closing this one.

**Comment by MTRNord** [comment]
Thanks :)
